### ISSUE TITLE
notEmpty() not working

### ISSUE THREAD

**Issue by dhpradeep** [express-validator/express-validator#476]
Hi , i'm using latest version of express-validator, 
and there is nothing about notEmpty in readme.md .
when i use it shows this message
`check('topic', 'this is not be empty').notEmpty(),`

`TypeError: check(...).notEmpty is not a function`

so how can i use ?

**Comment by Ethan-ks** [comment]
I see that you closed this issues the same day you posted it, but I am now facing the same issue as you. It appears that "notEmpty()" is no longer provided by validator.js as a validator, but is there an alternative? I simply want to make sure that an input on a form is not empty. I realize this can be done with the "required" html attribute, but I want to know if there is a standard way to do it via express-validator, so I can keep all my validation in one place.

**Comment by gustavohenke** [comment]
Hi @Ethan-ks,
Looks like this issue fell off my radar.

> "notEmpty()" is no longer provided by validator.js as a validator

It was never a validator.js validator. Actually, it was an express-validator validator in the legacy API (in fact, [it still is](https://express-validator.github.io/docs/legacy-api.html#legacy-validation-chain)).

> is there an alternative?

Yes, with the check API you can use `check('something').not().isEmpty()` for now.

I acknowledge this was way too popular in previous versions of express-validator, so I'm willing to bring it back.

**Comment by Ethan-ks** [comment]
I think that `check('thing').not().isEmpty()` is actually plenty straightforward, the problem lies in the `check().not()` method not being clearly documented. I looked through the check API page as well as the list of validators withing the validator.js documentation, and didn't see any mention of it. If I'm simply missing it, please point it out to me, otherwise, I think mentioning the existance of the method would avoid future issues like this. Thanks.

**Comment by gustavohenke** [comment]
> I looked through the check API page as well as the list of validators withing the validator.js documentation, and didn't see any mention of it

Again, not a validator.js thing.

The check API returns a [Validation Chain](https://express-validator.github.io/docs/validation-chain-api.html), which in turn contains all of validator.js validators + a few other methods (these being documented there).

Do you know of any way we could improve the docs to make these more prominent?

**Comment by maryamkhidir** [comment]
Thanks @gustavohenke for the solution. I was facing a similar bug to @dhpradeep and I was able to fix it with the solution you provided. Thanks man

**Comment by shanutthankachan** [comment]
use, `check('lastName').not().isEmpty().withMessage('The Last name is required')`

**Comment by jakazzy** [comment]
Thanks @gustavohenke I faced this issue. Followed your solution and it worked. Big ups!

**Comment by jaguardev** [comment]
using not().isEmpty() works  for imperative style great
what about schema validation?
let's say how to validate not empty and not a number?

**Comment by gustavohenke** [comment]
Hey @jaguardev,
You would do it like this:

```js
checkSchema({
	id: {
		isEmpty: {
			negated: true,
		}
	}
})
```

**Comment by gustavohenke** [comment]
Shipped in v6.3.0!
